### ISSUE TITLE
A false positive about the document case

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11197]
I have read check documentation: https://checkstyle.sourceforge.io/config_naming.html#ClassTypeParameterName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
    
<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="ClassTypeParameterName">
      <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
    </module>
  </module>
</module>


/var/tmp $ cat Test.java
class MyClass1<T> {}        // violation
class MyClass2<t> {}        // violation
class MyClass3<abc> {}      // violation
class MyClass4<LISTENER> {} // violation
class MyClass5<RequestT> {} // OK

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/vanguard/projects/SAMutator/tools/Test.java:2:16: Name 't' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
[ERROR] /home/vanguard/projects/SAMutator/tools/Test.java:3:16: Name 'abc' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
[ERROR] /home/vanguard/projects/SAMutator/tools/Test.java:4:16: Name 'LISTENER' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
Audit done.
Checkstyle ends with 3 errors.
```
---
**Describe what you expect in detail.**
---
Actually, this is a document case, it is the last example of [Examples](https://checkstyle.sourceforge.io/config_naming.html#ClassTypeParameterName_Examples). The document has been tagged 4 violations, but actually Checkstyle can only detect 3 violations, except the line 1. Thanks.

**Comment by nmancus1** [comment]
Thanks for pointing this out, example should read as follows:

```
class MyClass1<T> {}        // ok
class MyClass2<t> {}        // violation
class MyClass3<abc> {}      // violation
class MyClass4<LISTENER> {} // violation
class MyClass5<RequestT> {} // OK

```

This config is used to conform to [Google's style guide](https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s5.2.8-type-variable-names). While fixing this issue, we could add an example like `class MyClass6<T2> {}        // ok` to show other acceptable naming per Google style guide.

**Comment by Kevin222004** [comment]
i am on it

**Comment by Kevin222004** [comment]
@nrmancuso @romani I don't see any issue in this only docs need to be updated as per `(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)`
single T is excepted

**Comment by nrmancuso** [comment]
@Kevin222004 yes, issue is for documentation update

**Comment by Atharv3221** [comment]
No inconsistency in current  [examples](https://checkstyle.sourceforge.io/checks/naming/classtypeparametername.html#Examples) this issue can be closed

Example 1
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassTypeParameterName"/>
  </module>
</module>
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Example1.java
class Example1 {
  class MyClass1<T> {}
  class MyClass2<t> {}        // violation 'Name 't' must match pattern'
  class MyClass3<abc> {}      // violation 'Name 'abc' must match pattern'
  class MyClass4<LISTENER> {} // violation 'Name 'LISTENER' must match pattern'
  class MyClass5<RequestT> {} // violation 'Name 'RequestT' must match pattern'
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.4.2-all.jar -c config.xml Example1.java
Starting audit...
[ERROR] /home/atharv/test/Example1.java:3:18: Name 't' must match pattern '^[A-Z]$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example1.java:4:18: Name 'abc' must match pattern '^[A-Z]$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example1.java:5:18: Name 'LISTENER' must match pattern '^[A-Z]$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example1.java:6:18: Name 'RequestT' must match pattern '^[A-Z]$'. [ClassTypeParameterName]
Audit done.
Checkstyle ends with 4 errors.
```

Example 2

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config2.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassTypeParameterName">
      <property name="format" value="^[A-Z]{2,}$"/>
    </module>
  </module>
</module>
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Example2.java
class Example2 {
  class MyClass1<T> {}        // violation 'Name 'T' must match pattern'
  class MyClass2<t> {}        // violation 'Name 't' must match pattern'
  class MyClass3<abc> {}      // violation 'Name 'abc' must match pattern'
  class MyClass4<LISTENER> {}
  class MyClass5<RequestT> {} // violation 'Name 'RequestT' must match pattern'
}
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.4.2-all.jar -c config2.xml Example2.java
Starting audit...
[ERROR] /home/atharv/test/Example2.java:2:18: Name 'T' must match pattern '^[A-Z]{2,}$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example2.java:3:18: Name 't' must match pattern '^[A-Z]{2,}$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example2.java:4:18: Name 'abc' must match pattern '^[A-Z]{2,}$'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example2.java:6:18: Name 'RequestT' must match pattern '^[A-Z]{2,}$'. [ClassTypeParameterName]
Audit done.
Checkstyle ends with 4 errors.
```

Example 3

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat config3.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassTypeParameterName">
      <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
    </module>
  </module>
</module>
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Example3.java
class Example3 {
  class MyClass1<T> {}
  class MyClass2<t> {}        // violation 'Name 't' must match pattern'
  class MyClass3<abc> {}      // violation 'Name 'abc' must match pattern'
  class MyClass4<LISTENER> {} // violation 'Name 'LISTENER' must match pattern'
  class MyClass5<RequestT> {}
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-13.4.2-all.jar -c config3.xml Example3.java
Starting audit...
[ERROR] /home/atharv/test/Example3.java:3:18: Name 't' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example3.java:4:18: Name 'abc' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
[ERROR] /home/atharv/test/Example3.java:5:18: Name 'LISTENER' must match pattern '(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)'. [ClassTypeParameterName]
Audit done.
Checkstyle ends with 3 errors.
```
